src/bin/wasm2obj.rs

```diff
@@ -63,7 +63,7 @@ The translation is dependent on the environment chosen.
 The default is a dummy environment that produces placeholder values.
 
 Usage:
-    wasm2obj [--target TARGET] [-Odg] [--disable-cache | --cache-config=<cache_config_file>] [--enable-simd] <file> -o <output>
+    wasm2obj [--target TARGET] [-Odg] [--disable-cache | --cache-config=<cache_config_file>] [--enable-simd] [--always-lightbeam | --always-cranelift] <file> -o <output>
```

Comment:
Why the need for the prefix of "always" here and and in other places. Also, if cranelift if default, what impact does --always-cranelift have as a flag?

Reply:
"always" is in anticipation of having a tiered mode using Lightbeam and Cranelift and perhaps other compilers in combination. I could be convinced it's too verbose though :-).

Cranelift is the default today, but in the future, we may change the default. --always-cranelift is a way to ask for Cranelift without depending on what the default is.

Reply:
Ok .. cool. I was going to amend my comment because I was thinking as much but left it alone. One caveat I was thinking is it were in anticipation of tiered it would apply for the jit but not the wasm2obj. That said, I do think for consistency sake having always_* for wasm2obj is also reasonable.

Reply:
> "always" is in anticipation of having a tiered mode using Lightbeam and Cranelift and perhaps other compilers in combination. I could be convinced it's too verbose though :-).
> 
> Cranelift is the default today, but in the future, we may change the default. --always-cranelift is a way to ask for Cranelift without depending on what the default is.

Do you have some plan in mind w.r.t. a tiered system? Workloads and use cases being targeted?

Reply:
Yeah, I agree about wasm2obj. It may never support tiering, but we should keep the command-line options consistent.

The main use case for tiering is fast startup time for the first run of a module, which is important for any use case where end users are running code they just downloaded. Subsequent runs of a module can use the compilation cache (now enabled on master!), but that doesn't help make a good first impression :-).

Thinking about this more, I convinced myself that the "always" part isn't really necessary. We can shorten these options to `--cranelift` and `--lightbeam` without loss of generality.